generator_process/actions/prompt_to_image.py

```diff
@@ -323,17 +351,23 @@ def approximate_decoded_latents(latents):
 
 def model_snapshot_folder(model, preferred_revision: str | None = None):
     """ Try to find the preferred revision, but fallback to another revision if necessary. """
-    storage_folder = model
+    import diffusers
+    storage_folder = os.path.join(diffusers.utils.DIFFUSERS_CACHE, model)
     if os.path.exists(os.path.join(storage_folder, 'model_index.json')): # converted model
         snapshot_folder = storage_folder
     else: # hub model
-        ref_path = None
-        for revision in os.listdir(os.path.join(storage_folder, "refs")):
-            ref_path = os.path.join(storage_folder, "refs", revision)
-            if revision == preferred_revision:
-                break
-        if ref_path is None:
+        revisions = os.listdir(os.path.join(storage_folder, "refs"))
+        if len(revisions) == 0:
             return None
+        elif preferred_revision in revisions:
+            revision = preferred_revision
+        elif preferred_revision in [None, "fp16"] and "main" in revisions:
```

Comment:
Could you also add a check like the one that takes place below to make sure the revision has all of the necessary files. I had a case where the `main` revision was there but only had the model index.

Something like this should work:

```python
def revision_is_valid(revision):
    ref_path = os.path.join(storage_folder, "refs", revision)
    with open(ref_path) as f:
        commit_hash = f.read()

    snapshot_folder = os.path.join(storage_folder, "snapshots", commit_hash)
    return len(os.listdir(snapshot_folder)) > 1
```